controllers/destination_controller.go

```diff
@@ -178,11 +179,11 @@ func (r *DestinationReconciler) createDependenciesPathWithExample(writer writers
 func (r *DestinationReconciler) deleteDestination(o opts, destination *v1alpha1.Destination, writer writers.StateStoreWriter) (ctrl.Result, error) {
 	if controllerutil.ContainsFinalizer(destination, destinationCleanupFinalizer) {
 		if success, err := r.deleteDestinationWorkplacements(o, destination); !success || err != nil {
-			return defaultRequeue, nil
+			return defaultRequeue, err
```

Comment:
This didn't cause tests to fail, but may still not be correct

Reply:
Is it worth no-linting with a comment for now?